### PR TITLE
docs: adds documentation for autogenerated events

### PR DESCRIPTION
**Pull request by paschalidi**
Adds documentation for the https://jira.dhis2.org/browse/DHIS2-10315

Design doc that I used for inspiration is https://docs.google.com/document/d/1E5dBP5_BZZ0pLt573Xd0GQB6kve2WUpNdoew3kCFnUo/edit?pli=1#heading=h.e4oyflha5yjy

### CODE REVIEW

File: src/commonmark/en/content/user/using-the-capture-app.md [hunk 1]
```diff
@@ -183,6 +183,63 @@ The second option, is to register a tracked entity instance with program and enr
...
+A program can be configured to have zero or more program stages which are automatically generated upon a new enrollment. 
+These stages will be auto generated based on the metadata configuration, as explained below.
+
+To configure the auto generation of an event you need to take the following steps. 
+1. Open the maintenance app
+
+2. Select the Program tab
+![](resources/images/capture_app/auto-generated-01.png)
+
+3. Select a Tracker program.
```
Comment (paschalidi) [inline src/commonmark/en/content/user/using-the-capture-app.md]:
```suggestion
3. Select a Tracker program
```

File: src/commonmark/en/content/user/using-the-capture-app.md [hunk 2]
```diff
@@ -183,6 +183,63 @@ The second option, is to register a tracked entity instance with program and enr
...
+5. Click on the stage you would like to configure
+![](resources/images/capture_app/auto-generated-04.png)
+
+6. Mark the stage as Auto-generated
+![](resources/images/capture_app/auto-generated-05.png)
+
+Now, for every new enrollment in this program one event will be auto generated. One program can also have multiple stages marked as auto generated.
+For all the auto generated events 
+
+ a) the organisation unit will be the same as the user is reporting for during the enrollment and
```
Comment (paschalidi) [inline src/commonmark/en/content/user/using-the-capture-app.md]:
```suggestion
 a) the organisation unit will be the same as the user is reporting for, during the enrollment and 
```

File: src/commonmark/en/content/user/using-the-capture-app.md [hunk 3]
```diff
@@ -183,6 +183,63 @@ The second option, is to register a tracked entity instance with program and enr
...
+
+6. Mark the stage as Auto-generated
+![](resources/images/capture_app/auto-generated-05.png)
+
+Now, for every new enrollment in this program one event will be auto generated. One program can also have multiple stages marked as auto generated.
+For all the auto generated events 
+
+ a) the organisation unit will be the same as the user is reporting for, during the enrollment and 
+ 
+ b) all these events will be part of the current enrollment.
```
Comment (paschalidi) [inline src/commonmark/en/content/user/using-the-capture-app.md]:
```suggestion
 b) all the events will be part of the current enrollment. 
```

File: src/commonmark/en/content/user/using-the-capture-app.md [hunk 4]
```diff
@@ -183,6 +183,63 @@ The second option, is to register a tracked entity instance with program and enr
...
+
+ a) the organisation unit will be the same as the user is reporting for, during the enrollment and 
+ 
+ b) all the events will be part of the current enrollment. 
+ 
+Based on configuration, the status of the auto generated event can either be ACTIVE or SCHEDULE.
+
+#### Active type of event
+
+If the stage has the "Open data entry form after enrollment" selected, then the event will be generated into the ACTIVE status, and an execution date will be calculated for the event, in addition to a due date.
```
Comment (paschalidi) [inline src/commonmark/en/content/user/using-the-capture-app.md]:
```suggestion
If the stage has the "Open data entry form after enrollment" selected, then the event will be generated into the ACTIVE status. Also its execution date will be calculated for the event, in addition to a due date.
```

File: src/commonmark/en/content/user/using-the-capture-app.md [hunk 5]
```diff
@@ -183,6 +183,63 @@ The second option, is to register a tracked entity instance with program and enr
...
+The generation happens based on either the enrollment date or the incident date. You can choose the reporting date from the dropdown menu "Report date to use".
+![](resources/images/capture_app/auto-generated-06.png)
+
+As shown in the image you have three options, a) Incident date b) Enrollment date or c) No value. 
+Choosing reporting date as "Incident date" indicates that both the event execution date and due date will be the same as the incident date.
+Choosing reporting date as either "Enrollment date" or "No value" indicates that both the event execution date and due date will be the same as the enrollment date.
+
+#### Schedule type of event
+
+When the “Open data entry after enrollment” is not checked, it means that the event generated should be a SCHEDULE event.
```
Comment (paschalidi) [inline src/commonmark/en/content/user/using-the-capture-app.md]:
```suggestion
When the "Open data entry after enrollment" is not checked, it means that the event generated should be a SCHEDULE event. 
```

Comment (paschalidi) [inline src/commonmark/en/content/user/using-the-capture-app.md]:
```suggestion
When the “Open data entry after enrollment” is not checked, it means that the event generated will be a SCHEDULE event. 
```

File: src/commonmark/en/content/user/using-the-capture-app.md [hunk 6]
```diff
@@ -183,6 +183,63 @@ The second option, is to register a tracked entity instance with program and enr
...
+The generation happens based on either the enrollment date or the incident date. You can choose the reporting date from the dropdown menu "Report date to use".
+![](resources/images/capture_app/auto-generated-06.png)
+
+As shown in the image you have three options, a) Incident date b) Enrollment date or c) No value. 
+Choosing reporting date as "Incident date" indicates that both the event execution date and due date will be the same as the incident date.
+Choosing reporting date as either "Enrollment date" or "No value" indicates that both the event execution date and due date will be the same as the enrollment date.
+
+#### Schedule type of event
+
+When the "Open data entry after enrollment" is not checked, it means that the event generated should be a SCHEDULE event.
```
Comment (paschalidi) [inline src/commonmark/en/content/user/using-the-capture-app.md]:
```suggestion
When the "Open data entry after enrollment" is not checked, it means that the event generated will be a SCHEDULE event. 
```

File: src/commonmark/en/content/user/using-the-capture-app.md [hunk 7]
```diff
@@ -183,6 +183,63 @@ The second option, is to register a tracked entity instance with program and enr
...
+The scheduled event does not have an execution date, but only a due date. The due date for these future events are calculated based on either enrollment date or incident date. If the flag below is checked, the reference date is the enrollment date, if the flag is not checked, the incident date is used.
+![](resources/images/capture_app/auto-generated-07.png)
+
+When there is no incident date, the reference date will fall back on the enrollment date regardless of whether the flag above is checked.
+
+On SCHEDULE type of events the user can also configure the "Scheduled days from start". Which means if a stage has a number in "Scheduled days from start" the reference date will increased by that number. 
+In the example below we increase the due date by 30 days.
+![](resources/images/capture_app/auto-generated-08.png)
+
+When the standard interval does not contain a number or contains 0 the reference date is used without adding any days to it.
```
Comment (paschalidi) [inline src/commonmark/en/content/user/using-the-capture-app.md]:
```suggestion
When the "Scheduled days from start" does not contain a number or contains 0 the reference date is used without adding any days to it.
```

**Review by geethaalwan — COMMENTED** [review]
All the implemented functionality are added in the document. Looks good to me.

**Review by vilkg — APPROVED** [review]

**Pull request merged**